ksql-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestCase.java

```diff
@@ -326,24 +329,50 @@ static void processSingleRecord(
 
     final Topic sinkTopic = testDriver.getSinkTopic();
 
+    processRecordsForTopic(
```

Comment:
why is this a special case? (i.e. why not just have this be in the for loop below by removing the if clause?)

Reply:
The `possibleSinkTopics` comes from the provided input (such as changelog topics) and is optional so it may or may not include the final sink topic.